### PR TITLE
feat: Add activations concept doc

### PR DESCRIPTION
**Pull request by zmitchell**
This also adds the D2 plugin for MkDocs so that we can use D2 to make diagrams. D2 is just generally a nicer language than Mermaid and renders better looking diagrams out of the box.

### CODE REVIEW

**Comment by zmitchell** [comment]
Totally forgot to discuss runtime vs. develop mode, so I'll add that before merging

**Comment by jennymahmoudi** [comment]
good catch!

File: docs/concepts/activation.md
Comment (mkenigs) [inline docs/concepts/activation.md:56]: nit: I think 1st person sounds a little out of place here since it's docs not a blog

File: docs/concepts/activation.md
Comment (mkenigs) [inline docs/concepts/activation.md:288]: nit: I think "Somewhere" is confusing and if we just say save environment variables" that's probably enough information. I don't think we need a box that basically means "you don't care" because...they don't care.
```suggestion
files: \"Somewhere\"
```

File: docs/concepts/activation.md
Comment (mkenigs) [inline docs/concepts/activation.md:154]: I like this. I think this page gives people the information they need to really understand the details of what's going on if they want, but it offers that information in a way that doesn't get in the way of people who just need a surface level understanding

**Review by mkenigs — APPROVED** [review]
I'll probably merge this before the release and if we want any followup we can add that later

**Pull request merged**
